presto-spi/src/main/java/com/facebook/presto/spi/plan/WindowNode.java

```diff
@@ -85,34 +85,33 @@ public WindowNode(
         checkArgument(specification.getPartitionBy().containsAll(prePartitionedInputs), "prePartitionedInputs must be contained in partitionBy");
         Optional<OrderingScheme> orderingScheme = specification.getOrderingScheme();
         checkArgument(preSortedOrderPrefix == 0 || (orderingScheme.isPresent() && preSortedOrderPrefix <= orderingScheme.get().getOrderByVariables().size()), "Cannot have sorted more symbols than those requested");
-        checkArgument(preSortedOrderPrefix == 0 || ImmutableSet.copyOf(prePartitionedInputs).equals(ImmutableSet.copyOf(specification.getPartitionBy())), "preSortedOrderPrefix can only be greater than zero if all partition symbols are pre-partitioned");
+        checkArgument(preSortedOrderPrefix == 0 || prePartitionedInputs.equals(new HashSet<>(specification.getPartitionBy())), "preSortedOrderPrefix can only be greater than zero if all partition symbols are pre-partitioned");
 
         this.source = source;
-        this.prePartitionedInputs = ImmutableSet.copyOf(prePartitionedInputs);
+        this.prePartitionedInputs = unmodifiableSet(new HashSet<>(prePartitionedInputs));
         this.specification = specification;
-        this.windowFunctions = ImmutableMap.copyOf(windowFunctions);
+        this.windowFunctions = unmodifiableMap(new HashMap<>(windowFunctions));
         this.hashVariable = hashVariable;
         this.preSortedOrderPrefix = preSortedOrderPrefix;
     }
 
     @Override
     public List<PlanNode> getSources()
     {
-        return ImmutableList.of(source);
+        return singletonList(source);
     }
 
     @Override
     public List<VariableReferenceExpression> getOutputVariables()
     {
-        return ImmutableList.<VariableReferenceExpression>builder()
-                .addAll(source.getOutputVariables())
-                .addAll(windowFunctions.keySet())
-                .build();
+        List<VariableReferenceExpression> outputVariables = new ArrayList<>(source.getOutputVariables());
+        outputVariables.addAll(windowFunctions.keySet());
+        return unmodifiableList(outputVariables);
     }
 
     public Set<VariableReferenceExpression> getCreatedVariable()
     {
-        return windowFunctions.keySet();
+        return unmodifiableSet(windowFunctions.keySet());
```

Comment:
Why we need change here?

Reply:
you're right, not needed. I wasn't sure if the set view of the unmodifiable map would also be unmodifiable, but it is.